app/models/ckey2bibframe.rb

```diff
@@ -0,0 +1,42 @@
+# SSH to Symphony and run a catalogdump on a ckey
+class Ckey2bibframe
+  include OperatingSystem
+  include ActiveModel::Model
+  include ActiveModel::Validations
+  attr_accessor :ckey, :baseuri, :marcxml
+  validates :ckey, numericality: { message: ': You must enter a known ckey!' }
+
+  def convert
+    symphony_ssh = "ssh #{Settings.symphony_user}@#{Settings.symphony_host}"
+    symphony_env = "source #{Settings.symphony_env}"
+    catalogdump = "echo #{ckey} | #{Settings.symphony_catalogdump}"
+
+    jar = 'lib/xform-marc21-to-xml-jar-with-dependencies.jar'
+    java_command = "java -Djava.security.egd=file:/dev/../dev/urandom -cp #{jar} edu.stanford.MarcToXMLStream"
+
+    "#{symphony_ssh}" " '#{symphony_env} && #{catalogdump}' | " "#{java_command}"
+  end
+
+  def marc21_to_xml
+    begin
+      marcxml = execute_command(convert).force_encoding('UTF-8')
+    rescue => e
+      Rails.logger.warn('Unable to connect to Symphony via SSH.')
+      Rails.logger.error(e)
+    end
+    File.write(marcxml_file, marcxml)
+    Nokogiri::XML(marcxml, &:noblanks)
+  end
+
+  def marcxml_to_bibframe
+    bf2_xsl = 'loc_marc2bibframe2/xsl/marc2bibframe2.xsl'
+    command = "xsltproc --stringparam baseuri #{baseuri} #{bf2_xsl} #{marcxml_file}"
+    bibframe = execute_command(command).force_encoding('UTF-8')
+    FileUtils.rm(marcxml_file)
+    Nokogiri::XML(bibframe, &:noblanks)
+  end
```

Comment:
The `bf2_xsl` string can be a class constant (possibly a settings constant, in case they ever change where it is located in their repo or in case the deployment system changes where it is installed).